oarepo_oidc_einfra/resources.py

```diff
@@ -67,18 +89,94 @@ def upload_dump(self):
+        and use it to accept the invitation.
 
-def create_rest_blueprint(app):
+        Note:
+        If user accepts the invitation but this endpoint is not called, the invitation will be forever
+        in the submitted state (until expiration). The user will still be able to access the community
+        because the AAI will return the correct capabilities for the user.
+
+        Currently, the PERUN api does not return the ID of the created invitation, so we cannot store it
+        and check in a background task if the invitation was accepted and then change the state of the request.
+
+        """
+        assert request.view_args is not None
+
+        request_id = decrypt(request.view_args["request_id"])
+
+        # get the invitation request and check if it is submitted.
+        invitation_request = Request.get_record(request_id)
+        assert invitation_request.state == "submitted"
+
+        # if its user is not the current user, we might to delete
+        # the user on the request so that it does not pollute the space
+        request_user_id = invitation_request.payload.get("user_id")
+        if request_user_id != str(g.identity.id):
+            user = User.query.filter_by(User.id == request_user_id).one()
+            if not user.is_active:
+                db.session.delete(user)
+            else:
+                log.error(
+                    "Invitation check failed: The user for which the invitation was sent (%s) "
+                    "is an active user and is not the same as the current user %s, thus the "
+                    "invitation was not accepted. This means that we need to check the users if duplicity"
+                    "exists and if so, we need to merge them somehow.",
+                    request_user_id,
+                    g.identity.id,
+                )
+                raise PermissionDenied(
+                    "The invitation was intended for a different user"
+                )
+
+        # now, change the receiver to the current user
+        invitation_request.receiver = {"user": g.identity.id}
+        invitation_request.commit()
+
+        current_requests_service.execute_action(system_identity, request_id, "accept")
+        return redirect("/")
+
+
+def store_dump(request_data: bytes) -> str:
+    """Store the dump in the configured location and return the path.
+
+    The dump is stored in the bucket configured in the EINFRA_USER_DUMP_S3_BUCKET,
+    the actual path is put into the cache under the key EINFRA_LAST_DUMP_PATH
+    and the path is returned.
+
+    Storing the path into the cache means that even if the background task process
+    multiple dumps out of order, the last one will be always the one that is processed -
+    the previous ones will be ignored.
+    """
+    now = datetime.now(UTC).strftime("%Y-%m-%d-%H-%M-%S")
+    dump_path = f"{now}.json"
+    client = boto3.client(
```

Comment:
Store content checksum somewhere (object metadata), validate it when importing (protect against external tampering with dump data between store/import)

Reply:
ok